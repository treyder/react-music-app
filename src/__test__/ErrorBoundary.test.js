import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";
import ErrorBoundary from '../ErrorBoundary';

describe("ErrorBoundary", () => {
    let mountedErrorBoundary;
    let initialText = "<p>some text</p>";
    const errorBoundary = () => {
        if (!mountedErrorBoundary) {
            mountedErrorBoundary = shallow(
                <ErrorBoundary children={initialText}/>
            );
        }
        return mountedErrorBoundary;
    }

    beforeEach(() => {
        mountedErrorBoundary = undefined;
    });

    it('renders no error by default', () => {
        const tree = renderer.create(
          <ErrorBoundary children="<p>some text</p>"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders error on exception", () => {
        errorBoundary().instance().componentDidCatch("some error", "");
        errorBoundary().update();
        expect(errorBoundary().html()).toEqual("<h1>Something went wrong officer ...</h1>");
    });
});
