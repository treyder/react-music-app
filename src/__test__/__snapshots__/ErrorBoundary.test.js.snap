// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders no error by default 1`] = `"<p>some text</p>"`;
